src/Streamly/Internal/Data/Stream/IsStream/Eliminate.hs

```diff
@@ -228,7 +230,7 @@ import Prelude hiding
 -- @since 0.1.0
 {-# INLINE uncons #-}
 uncons :: (IsStream t, Monad m) => SerialT m a -> m (Maybe (a, t m a))
-uncons (SerialT m) = fmap (fmap (fmap IsStream.fromStream)) $ K.uncons m
+uncons m = fmap (fmap (fmap IsStream.fromStream)) $ K.uncons $ Stream.toStreamK m
```

Comment:
Stream.uncons

Reply:
This returns `t m a`, so can't be Stream.uncons.